### PR TITLE
Add integration test for Catalog V2 cluster repos

### PR DESCRIPTION
**Pull request by PennyScissors**
Issue: https://github.com/rancher/qa-tasks/issues/397, https://github.com/rancher/qa-tasks/issues/396, https://github.com/rancher/rancher/issues/38793

Add test to validate create/update/delete of HTTP and Git cluster repos

### CODE REVIEW

**Review by igomez06 — COMMENTED** [review]
One small comment, otherwise looks great!

File: tests/v2/integration/catalogv2/cluster_repo_test.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	ClusterRepoType = "catalog.cattle.io.clusterrepo"
```
[the remark this answers is not in the document]
Comment (igomez06) [reply tests/v2/integration/catalogv2/cluster_repo_test.go:19]: And just follow the naming convention we have for the other Steve resource types

**Review by igomez06 — APPROVED** [review]
LGTM

**Review by ronhorton — APPROVED** [review]

**Review by aiyengar2 — APPROVED** [review]

**Comment by igomez06** [comment]
I forgot to ask, but do we need a backport for v2.6 as well? @PennyScissors

**Comment by bmdepesa** [comment]
Yes, tests should be backported to v2.6 unless they are specific to 2.7x+ functionality.

**Comment by PennyScissors** [comment]
Alright, I'll backport this

**Pull request merged**